### PR TITLE
Create and store in the database the history of executed scheduled jobs

### PR DESCRIPTION
**Pull request by adityasrini**
Fixes issue #657

### CODE REVIEW

**Comment by steve-todorov** [comment]
@sbespalov could you please have a look at this and let @adityasrini know if you have any remarks?

**Comment by sbespalov** [comment]
@adityasrini is this PS assumed to be complete? I am asking becasue I can't find any logic which manages  `ScheduledTaskEntry`.

File: strongbox-cron/strongbox-cron-api/src/main/resources/db.changelog/v1.0.0/v1.0.0.11__ScheduledTaskEntry.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
```
Comment (sbespalov) [inline strongbox-cron/strongbox-cron-api/src/main/resources/db.changelog/v1.0.0/v1.0.0.11__ScheduledTaskEntry.xml]: changelog files should be placed in `data-service-config` module

**Comment by carlspring** [comment]
@adityasrini .

What is the status of this PR?

**Comment by adityasrini** [comment]
I still have more to go. I haven't added logic to handle `ScheduledTaskEntry` yet and since I need some guidance on that so I PM'ed sbespalov.

**Comment by adityasrini** [comment]
@sbespalov, I have a quick question. Is this how I should design and implement the ScheduledTaskEntryService interface?
https://gist.github.com/adityasrini/e7bb95fb4fa805c7964f118bb189f04b

**Comment by adityasrini** [comment]
Cross posting a comment I left on gist since gist doesn't send any notifications for activity.

 > Yeah I don’t think I’d need all. I was just going by how ArtifactEntry services were implemented which includes a bunch of methods that make sense for retrieving artifacts and not necessarily scheduled tasks. Just wanted to clarify this for that reason!

**Comment by adityasrini** [comment]
@sbespalov is it safe to assume that the task names have a one to many relationship with thread names? Like one task can run on many threads.

**Comment by adityasrini** [comment]
Because if there is a one to many relationship anywhere, I can implement the necessary methods to handle that and use a `Set` instead of a `String` for thread name.

**Comment by sbespalov** [comment]
@adityasrini I think you don't need any Entity relations here, `ScheduledTaskEntry` probably should be just plane entry with only thread name.

sorry for late response :)

**Comment by andreybleme** [comment]
Hi, @adityasrini and @sbespalov what is avoiding this code to be merged?

**Comment by andreybleme** [comment]
@carlspring I'll wait for @adityasrini 's response :)

**Comment by adityasrini** [comment]
@andreybleme have at it :)

**Comment by adityasrini** [comment]
@carlspring shall I close this PR since @andreybleme will be picking this up?

**Comment by carlspring** [comment]
Hi @adityasrini ,

Thanks for replying! Yeah, I think it would be best to close and restart it from scratch. Thanks for effort!

**Pull request closed without merging**